mesh/mesh.cpp

```diff
@@ -1119,6 +1119,100 @@ int Mesh::GetBdrFace(int BdrElemNo) const
    return fn;
 }
 
+Mesh::FaceInformation Mesh::GetFaceInformation(int f) const
+{
+   FaceInformation face;
+   int e1, e2;
+   int inf1, inf2;
+   int ncface;
+   GetFaceElements(f, &e1, &e2);
+   GetFaceInfos(f, &inf1, &inf2, &ncface);
+   face.elem_1_index = e1;
+   face.elem_1_orientation = inf1%64;
+   face.elem_1_local_face =inf1/64;
+   face.elem_2_local_face = inf2/64;
+   face.ncface = ncface;
+   face.point_matrix = nullptr;
+   // The following figures out face.location, face.conformity,
+   // face.elem_2_index, and face.elem_2_orientation.
+   if (f < GetNumFaces()) // Non-ghost face
+   {
+      if (e2>=0)
+      {
+         face.location = FaceLocation::Local;
+         face.elem_2_index = e2;
+         if (ncface==-1)
+         {
+            face.conformity = FaceConformity::Conforming;
+            face.elem_2_orientation = inf2%64;
+         }
+         else // ncface >= 0
+         {
+            face.conformity = FaceConformity::NonConformingSlave;
+            MFEM_ASSERT(inf2%64==0, "unexpected slave face orientation.");
+            face.elem_2_orientation = inf2%64;
+            face.point_matrix = nc_faces_info[ncface].PointMatrix;
+         }
+      }
+      else // e2<0
+      {
+         if (ncface==-1)
+         {
+            face.conformity = FaceConformity::Conforming;
+            if (inf2<0)
+            {
+               face.location = FaceLocation::Boundary;
+               face.elem_2_index = e2;
+               face.elem_2_orientation = -1;
+            }
+            else // inf2 >= 0
+            {
+               face.location = FaceLocation::Shared;
+               face.elem_2_index = -1 - e2;
+               face.elem_2_orientation = inf2%64;
+            }
+         }
+         else // ncface >= 0
+         {
+            if (inf2 < 0)
+            {
+               face.location = e2==-1 ? FaceLocation::Local : FaceLocation::Shared;
+               face.conformity = FaceConformity::NonConformingMaster;
+               face.elem_2_index = e2==-1 ? e2 : -1 - e2;
+               face.elem_2_orientation = inf2%64;
```

Comment:
@v-dobrev This case seems to be problematic for @dylan-copeland, and seems to return incorrect information, the documentation states:
```
   // - Elem2No < 0 --> local "boundary" face; can be one of:
   //    - NCFace >= 0 --> non-conforming face; can be one of:
   //       - Elem2Inf < 0 --> master non-conforming face, interior or shared;
   //         In this case, Elem2No is -1; see GenerateNCFaceInfo().
```
Do you know if this code decipher the face information correctly?

Reply:
For a shared master NC face, it is unclear how to determine if it is actually shared or on the boundary, since `Elem2No = -1`.